Add dates for implementation features.

emkornfield opened the pull request:
This adds dates to the data model for implementation features.

- It then uses those dates to add hyperlinks to implementation notes when available.
- It creates a new table summarizing implementation status per year based on dates.
- Added an entry to extending BYTE_STREAM_SPLIT to additional types.

The data here was backfilled with AI and spot-checked so it is possible some of it might be off.

emkornfield committed:
Add dates for implementation features.

This adds dates to the data model for implementation features.

- It then uses those dates to add hyperlinks to implementation notes
  when available.
- It creates a new table summarizing implementation status per year
  based on dates.
- Added an entry to extending BYTE_STREAM_SPLIT to additional types.

The data here was backfilled with AI and spot-checked so it is possible
some of it might be off.

emkornfield committed:
fix compilation

emkornfield commented:
Sample table (some data has been adjusted): <img width="1059" height="679" alt="image" src="https://github.com/user-attachments/assets/84060c0b-fc44-430e-956e-dd75a84b07e3" />

Note the X's have a tool-tip showing which features are missing.

emkornfield committed:
fix some data

emkornfield committed:
update note

RussellSpitzer commented:
I like how this looks but I definitely lack the expertise to determine the accuracy of the table as of now.

wgtmac commented on data/implementations/engines.yaml:
```diff
@@ -4,52 +4,195 @@
   docs_url: https://arrow.apache.org/docs/cpp/parquet.html
-  versions: {}
+  versions:
+    default: "15.0.0"
```
Why default is 15.0.0?

emkornfield replied:
I thought this was when most of the basics where implemented.  going back it looks like JSON wasn't implemented until 18.0.0 so updated to that.

wgtmac commented on data/implementations/engines.yaml:
```diff
@@ -4,52 +4,195 @@
   docs_url: null
-  versions: {}
+  versions:
+    default: "15.0.0"
```
Same question here

wgtmac commented on content/en/docs/File Format/implementationstatus.md:
```diff
@@ -12,7 +12,7 @@ of this page by opening an issue or submitting a pull request.
 ### Legend
 The value in each box means:
-* ✅: supported. Footnote added when support is partial.
+* ✅: supported. Footnote added when support is partial. When data is available links to release notes are provided on the implementing version.
```
```suggestion
* ✅: supported. Footnote added when support is partial. When data is available, links to release notes are provided on the implementing version.
```

wgtmac commented on layouts/shortcodes/implementation-status.html:
```diff
@@ -125,11 +240,206 @@ <h3 id="{{ $categoryId }}"><a href="#{{ $categoryId }}">{{ $category.name }}</a>
+  </thead>
+  <tbody>
+    {{- range $engines -}}
+      {{- $engine := . -}}
```
I found the rendered result for arrow-go is confusing but I find it hard to interpret this logic.

<img width="914" height="118" alt="image" src="https://github.com/user-attachments/assets/23c9d32d-d5b6-4461-ae5c-58b4ba6a27d2" />

emkornfield replied:
hmm, I thought I fixed this let me look into it.

emkornfield replied:
This was a data issue where 18.4 should have been 2025-07-21.  The code relies on release date for canonical values, and there where some other data issues that I cleaned up (go split off from Arrow mono-repo after 18.0.0 and is only up to 18.5.0).

emkornfield committed:
Update content/en/docs/File Format/implementationstatus.md

Co-authored-by: Gang Wu <ustcwg@gmail.com>

emkornfield committed:
fix data issues

emkornfield commented:
@wgtmac thanks for the review, I went back and cleaned up more data after another pass through and generally made the defaults more recent.  I also added a prior Java release and a default so Java support doesn't look as bad in terms of lag.

wgtmac commented:
Thanks for the update! Generally looks good to me with minor questions as below:

- Is `data/implementations/support/.parquet-java.yaml.swp` added by accident?
- Should we provide a guidance for people to update the implementation status page in the future?

emkornfield commented:
> Is data/implementations/support/.parquet-java.yaml.swp added by accident?

Yes, need to update the .gitignore for this repo, will do it a separate PR.

> Should we provide a guidance for people to update the implementation status page in the future?

Yes, I think this makes the most sense in the parquet-format CONTRIBUTING guide?  I'll open a separate PR there?

emkornfield committed:
remove swaps and add git ignore

emkornfield commented:
@wgtmac thanks for review, updated

wgtmac reviewed (approved):
Thanks a lot @emkornfield!

emkornfield commented:
Thanks for the review @wgtmac going to merge.
